airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/typing_deduping/JdbcTypingDedupingTest.kt

```diff
@@ -26,12 +31,20 @@ import org.jooq.impl.DSL
 abstract class JdbcTypingDedupingTest : BaseTypingDedupingTest() {
     protected var database: JdbcDatabase? = null
     private var dataSource: DataSource? = null
+    protected open val dialect: SQLDialect = SQLDialect.DEFAULT
+    protected val dslContext: DSLContext
```

Comment:
NIT: Why the need for the getter override rather than just setting the value of the field?

Reply:
it generates a warning. Not actually sure why, given that dialect is a `val` not a `var`... 
![image](https://github.com/airbytehq/airbyte/assets/5741425/bf91f77a-484f-4767-a220-832ca75a250d)

Reply:
yeah, I saw some similar errors and need to do some reading about

Reply:
I looked into it and it kinda makes sense: the super() constructor is always called before the child's constructor, So when the super constructr uses dialect, it's not set yet by the child instance. so a proper way to fix it is to set dialect as a constructor parameter, with a default value

Reply:
aha. that makes sense. I'll move it into the constructor. I think I can actually just kill the `dialect` field entirely, afaict it's only ever used to construct dslContext

(but also, ugh, my personal preference was to never mix field initializers with constructors. i.e. if a class had an explicit constructor, I avoided using field initializers for approximately this exact reason. Slightly annoying that it's the "proper" way in kotlin)